openpype/hosts/houdini/addon.py

```diff
@@ -4,44 +4,57 @@
 HOUDINI_HOST_DIR = os.path.dirname(os.path.abspath(__file__))
 
 
-class HoudiniAddon(OpenPypeModule, IHostAddon):
-    name = "houdini"
-    host_name = "houdini"
+def merge_paths(*paths):
+    """Merge path strings to a single uniqified `os.pathsep` joined string.
 
-    def initialize(self, module_settings):
-        self.enabled = True
+    Each path argument can itself be `os.pathsep` joined string.
+    >>> merge_paths("A", "A;B;C", "D;E")
+    "A;B;C;D;E"
 
-    def add_implementation_envs(self, env, _app):
-        # Add requirements to HOUDINI_PATH and HOUDINI_MENU_PATH
-        startup_path = os.path.join(HOUDINI_HOST_DIR, "startup")
-        new_houdini_path = [startup_path]
-        new_houdini_menu_path = [startup_path]
-
-        old_houdini_path = env.get("HOUDINI_PATH") or ""
-        old_houdini_menu_path = env.get("HOUDINI_MENU_PATH") or ""
+    Returns:
+        str: Single joined path using `os.pathsep`
 
-        for path in old_houdini_path.split(os.pathsep):
+    """
+    result = []
+    processed = set()
+    for paths_str in paths:
+        for path in paths_str.split(os.pathsep):
             if not path:
                 continue
 
-            norm_path = os.path.normpath(path)
-            if norm_path not in new_houdini_path:
-                new_houdini_path.append(norm_path)
-
-        for path in old_houdini_menu_path.split(os.pathsep):
-            if not path:
+            path = os.path.normpath(path)
+            if path in processed:
```

Comment:
Why not check against `result` directly? I'm not sure the `processed` set is necessary, something like:
```py
result = []

for paths_str in paths:
    for path in paths_str.split(os.pathsep):
        if not path:
            continue

        path = os.path.normpath(path)
	if path not in result:
            result.append(path)

return os.pathsep.join(result)
```

Reply:
Because that'd be way slower as a lookup. ;)